cluster-autoscaler/cloudprovider/clusterapi/clusterapi_scalableresource.go

```diff
@@ -60,3 +67,25 @@ type scalableResource interface {
 	InstanceGPUCapacity() (resource.Quantity, error)
 	InstanceMaxPodsCapacity() (resource.Quantity, error)
 }
+
+func unmarkMachineForDeletion(controller *machineController, machine *Machine) error {
+	u, err := controller.dynamicclient.Resource(*controller.machineResource).Namespace(machine.Namespace).Get(context.TODO(), machine.Name, metav1.GetOptions{})
```

Comment:
should i change these `context.TODO()` references to use the background context instead?

Reply:
context.TODO() is the same as context.Background, but is intended as a placeholder for a context that's passed in.

Reply:
ok, so TODO is fine for now since we don't have a context.

Reply:
If we intend to replace this with a proper context at some point (which I think we probably do given this is core code rather than test code, TODO is better IMO.

I've suggested we use Background in tests so that if anyone starts grepping for TODOs within our codebase, the tests don't flag false positives, we never reallly intend to use a real context in the tests but would likely want to for these requests